src/fair/simulation.py

```diff
@@ -27,6 +27,40 @@ class RenaissanceMan(SimulatedAgent):
     The Renaissance man prefers courses from multiple topics, up to a maximum quantity per topic.
     """
 
+    @staticmethod
```

Comment:
All good! I think it's a bit of a coin toss what's better in this case, since I need to reuse a lot of it but all the sampling goes away. I reverted it back to its own object for readability. Will work on the test case.